spec/controllers/books_controller_spec.rb

```diff
@@ -0,0 +1,93 @@
+require 'rails_helper'
+
+RSpec.describe BooksController do
+  before(:each) do 
+    user = FactoryGirl.create(:user)
+    sign_in user
+  end
+
+  let(:book) { FactoryGirl.create(:book) }
+  let(:author) { FactoryGirl.create(:author) }
+  describe 'GET index' do
+    context 'there are books' do
+      it 'populates an array of books' do
+        book
+        get :index
+        expect(assigns(:books)).to eq([book])
+      end
+
+      it 'renders the :index view' do
+        get :index
+        expect(response).to render_template('index')
+      end
+    end
+
+    context 'there are no books' do 
+      it 'populates an array of books' do
+        get :index
+        expect(assigns(:books)).to eq([])
+      end
+    end     
+  end
+
+  describe 'GET new' do
+    it 'populates an array of authors' do
+      author
+      get :new
+      expect(assigns(:authors)).to eq([author])
+    end
+
+    it 'renders the :new template' do
+      get :new
+      expect(response).to render_template('new')
+    end
+  end
+
+  describe 'POST create' do
+    context 'with valid attributes' do
+      let(:book_params) do
+        {
+          title: 'Prueba',
+          description: 'Descripcion',
+          author_id: FactoryGirl.create(:author).id,
```

Comment:
You can use `author` here instead of creating another one, you defined it above.